src/rez/package_order.py

```diff
@@ -126,21 +126,26 @@ def sort_key(self, package_name, version_like):
         """
         if isinstance(version_like, VersionRange):
             return tuple(self.sort_key(package_name, bound) for bound in version_like.bounds)
-        elif isinstance(version_like, _Bound):
+        if isinstance(version_like, _Bound):
             return (self.sort_key(package_name, version_like.lower),
                     self.sort_key(package_name, version_like.upper))
-        elif isinstance(version_like, _LowerBound):
+        if isinstance(version_like, _LowerBound):
             inclusion_key = -2 if version_like.inclusive else -1
             return self.sort_key(package_name, version_like.version), inclusion_key
-        elif isinstance(version_like, _UpperBound):
+        if isinstance(version_like, _UpperBound):
             inclusion_key = 2 if version_like.inclusive else 1
             return self.sort_key(package_name, version_like.version), inclusion_key
-        elif isinstance(version_like, Version):
+        if isinstance(version_like, Version):
             # finally, the bit that we actually use the sort_key_implementation for.
             return FallbackComparable(
                 self.sort_key_implementation(package_name, version_like), version_like)
-        else:
-            raise TypeError(version_like)
+        if version_like is None:
```

Comment:
@loonghao Can you adjust the type in the docstring to mention that None is a valid type please?